### PR TITLE
Bump googletest 1.8.0 -> 1.8.1 (take 2)

### LINKED ISSUE #35: Crash on build with clang 10 or later
osrf testing tools cpp builds fine with Clang 9 or earlier but breaks in Clang 10 with the below error.
I believe this is because of a bug in googletest fixed in 1.8.1 https://github.com/google/googletest/commit/13c5230bbf2bd3404e48b7aee33a9af2514d1b9a

```
--- stderr: osrf_testing_tools_cpp                                                                                                                                                                                                                     
In file included from /opt/ros/master/build/osrf_testing_tools_cpp/googletest-1.8.0-extracted/googletest-1.8.0-src/googletest/src/gtest-all.cc:39:
In file included from /opt/ros/master/build/osrf_testing_tools_cpp/googletest-1.8.0-extracted/googletest-1.8.0-src/googletest/include/gtest/gtest.h:62:
In file included from /opt/ros/master/build/osrf_testing_tools_cpp/googletest-1.8.0-extracted/googletest-1.8.0-src/googletest/include/gtest/gtest-param-test.h:193:
/opt/ros/master/build/osrf_testing_tools_cpp/googletest-1.8.0-extracted/googletest-1.8.0-src/googletest/include/gtest/internal/gtest-param-util-generated.h:107:8: error: definition of implicit copy constructor for 'ValueArray2<bool, bool>' is deprecated because it has a user-declared copy assignment operator [-Werror,-Wdeprecated-copy]
  void operator=(const ValueArray2& other);
       ^
/opt/ros/master/build/osrf_testing_tools_cpp/googletest-1.8.0-extracted/googletest-1.8.0-src/googletest/include/gtest/gtest-param-test.h:354:10: note: in implicit copy constructor for 'testing::internal::ValueArray2<bool, bool>' first required here
  return internal::ValueArray2<T1, T2>(v1, v2);
         ^
/opt/ros/master/build/osrf_testing_tools_cpp/googletest-1.8.0-extracted/googletest-1.8.0-src/googletest/include/gtest/gtest-param-test.h:1221:10: note: in instantiation of function template specialization 'testing::Values<bool, bool>' requested here
  return Values(false, true);
         ^
1 error generated.
make[2]: *** [googletest-1.8.0-extracted/googletest-1.8.0-build/googlemock/gtest/CMakeFiles/gtest.dir/src/gtest-all.cc.o] Error 1
make[1]: *** [googletest-1.8.0-extracted/googletest-1.8.0-build/googlemock/gtest/CMakeFiles/gtest.dir/all] Error 2
make[1]: *** Waiting for unfinished jobs....
make: *** [all] Error 2
---
```

### LINKED ISSUE #37: Nightly ASan and TSan builds are failing to build osrf_testing_tools_cpp
Hi, there have been failures in the nightly ASan and TSan builds due to a linker error caused while building this package. These failures started around the time https://github.com/osrf/osrf_testing_tools_cpp/pull/36 (gtest version bump) was merged.

All build details are in https://github.com/ros-tooling/aws-oncall/issues/83.

### PR DESCRIPTION
**Pull request by rotu**
Again try to bump googletest, but this time without breaking ASAN.
Changed how we link in the pthread library to be less fragile.

Includes changes from reverted PR #36
Fix #35 build under clang 10
Fix #37

### CODE REVIEW

**Comment by rotu** [comment]
@wjwwood, any objections? Should I change this to keep 1.8.0 and just put 1.8.1 in *next to* it?

File: osrf_testing_tools_cpp/cmake/osrf_testing_tools_cpp_extract_and_build_googletest.cmake [hunk 1]
```diff
@@ -72,9 +72,7 @@ macro(osrf_testing_tools_cpp_extract_and_build_googletest
     # (see https://wiki.gentoo.org/wiki/AddressSanitizer/Problems#pthread_linking_issues
     # for some additional information).  To work around that, we unconditionally
     # add the -pthread flag for Linux machines so it will always work
-    execute_process(COMMAND bash "-c" "sed -i 's@\${CMAKE_THREAD_LIBS_INIT}@\${CMAKE_THREAD_LIBS_INIT} -pthread@' ${__prefix}-src/googletest/cmake/internal_utils.cmake"
-      RESULT_VARIABLE result
-      WORKING_DIRECTORY ${CMAKE_BINARY_DIR}/${__prefix}-extracted)
+    link_libraries("-pthread")
```
Comment (wjwwood) [inline osrf_testing_tools_cpp/cmake/osrf_testing_tools_cpp_extract_and_build_googletest.cmake]: Hmm, seems like this change makes it no longer agree with the comments above?

@clalancette you added this logic to fix asan building in the past, can you comment?

Comment (clalancette) [reply osrf_testing_tools_cpp/cmake/osrf_testing_tools_cpp_extract_and_build_googletest.cmake]: > Hmm, seems like this change makes it no longer agree with the comments above?
> 
> @clalancette you added this logic to fix asan building in the past, can you comment?

It still agrees with the comment.  That is, asan still has the problem that it exposes a "fake" pthread that is good enough to fool CMake into thinking it found it, but it isn't good enough to successfully link.

What this change does is take advantage of the fact that this is a single CMake context, so using `link_libraries("-pthread")` works.  In my testing, this still links properly when compiling with asan, so :+1: from me.

Comment (wjwwood) [reply osrf_testing_tools_cpp/cmake/osrf_testing_tools_cpp_extract_and_build_googletest.cmake]: Ok, good enough for me. Thanks.

**Comment by wjwwood** [comment]
> Should I change this to keep 1.8.0 and just put 1.8.1 in next to it?

I think it's fine as-is.

CI:

* Linux [Build Status](http://ci.ros2.org/job/ci_linux/9463/)
* Linux-aarch64 [Build Status](http://ci.ros2.org/job/ci_linux-aarch64/5192/)
* macOS [Build Status](http://ci.ros2.org/job/ci_osx/7737/)
* Windows [Build Status](http://ci.ros2.org/job/ci_windows/9390/)
* Windows-container [Build Status](http://ci.ros2.org/job/ci_windows-container/230/)

**Comment by wjwwood** [comment]
I think the remaining test failures are known cases. Thanks for iterating on this @rotu!

**Pull request merged**